pkg/controller/servicemesh/member/controller.go

```diff
@@ -0,0 +1,242 @@
+		return []string{roll.Spec.ControlPlaneRef.Namespace}
+	})
+	if err != nil {
+		return err
+	}
+	// watch namespaces and trigger reconcile requests as those that match a member come and go
+	err = c.Watch(&source.Kind{Type: &corev1.Namespace{}}, &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(ns handler.MapObject) []reconcile.Request {
+			list := &v1.ServiceMeshMemberList{}
+			err := mgr.GetClient().List(context.TODO(), client.MatchingField("spec.controlPlaneRef.namespace", ns.Meta.GetName()), list)
+			if err != nil {
+				log.Error(err, "Could not list ServiceMeshMembers")
+			}
+
+			var requests []reconcile.Request
+			for _, smm := range list.Items {
+				requests = append(requests, reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name:      smm.Name,
+						Namespace: smm.Namespace,
+					},
+				})
+			}
+			return requests
+		}),
+	}, predicate.Funcs{
+		UpdateFunc: func(_ event.UpdateEvent) bool {
+			// we don't need to process the member on namespace updates
+			return false
+		},
+		DeleteFunc: func(_ event.DeleteEvent) bool {
+			// we don't need to process the member on namespace deletions
+			return false
+		},
+		GenericFunc: func(_ event.GenericEvent) bool {
+			// we don't need to process the member on generic namespace events
+			return false
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &MemberReconciler{}
+
+// MemberReconciler reconciles ServiceMeshMember objects
+type MemberReconciler struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	common.ResourceManager
+	scheme *runtime.Scheme
+}
+
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *MemberReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("ServiceMeshMember", request)
+	reqLogger.Info("Processing ServiceMeshMember")
+
+	member := &v1.ServiceMeshMember{}
+	err := r.Client.Get(context.TODO(), request.NamespacedName, member)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object
+		return reconcile.Result{}, err
+	}
+
+	memberRollKey := client.ObjectKey{
+		Name:      common.MemberRollName,
+		Namespace: member.Spec.ControlPlaneRef.Namespace,
+	}
+	memberRoll := &v1.ServiceMeshMemberRoll{}
+	isNewMemberRoll := false
+	err = r.Client.Get(context.TODO(), memberRollKey, memberRoll)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+		// MemberRoll doesn't exist, let's create it
+		isNewMemberRoll = true
+		memberRoll = &v1.ServiceMeshMemberRoll{
```

Comment:
No. The Member controller takes care of deleting the `SMMR`. This allows it to determine if the SMMR should even be deleted. Currently, it deletes it if it was the one who created it. But I'll need to change this, since it would also delete the SMMR in the following scenario (when it really shouldn't):
- user creates Member object
- controller creates SMMR
- user manually adds another namespace to the SMMR
- user deletes Member object

The controller shouldn't delete the SMMR in this case.